designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/component/package.scala

```diff
@@ -16,6 +21,20 @@ package object component {
 
   type NodeId = String
 
+  @ConfiguredJsonCodec sealed trait NodeMetadata {
+    def getNodeId: NodeId
```

Comment:
Regarding scala style, it should be just `def nodeId: Node`, to be easy to migrate between def/lazy val/val